File: base/src/main/java/com/mindera/skeletoid/routing/CloseActivityCommand.java
package com.mindera.skeletoid.routing;

import android.app.Activity;

/**
 * Closes the activity passed in the constructor
 */
public class CloseActivityCommand implements IRouteCommand {

    private final Activity mActivity;
    private int enterAnimation = 0;
    private int exitAnimation = 0;

    public CloseActivityCommand(Activity activity) {
        mActivity = activity;
    }

    public CloseActivityCommand(Activity activity, int enterAnimation, int exitAnimation) {
        this(activity);
        this.enterAnimation = enterAnimation;
        this.exitAnimation = exitAnimation;
    }

    @Override
    public void navigate() {
        mActivity.finish();
        if(enterAnimation != 0 || exitAnimation != 0) {
            mActivity.overridePendingTransition(enterAnimation, exitAnimation);
        }
    }
}

File: base/src/main/java/com/mindera/skeletoid/routing/CloseActivityCommand.kt
package com.mindera.skeletoid.routing

import android.app.Activity
import com.mindera.skeletoid.routing.interfaces.IRouteCommand

/**
 * Closes the activity passed in the constructor
 */
class CloseActivityCommand @JvmOverloads constructor(private val activity: Activity, private val enterAnimation: Int = 0, private val exitAnimation: Int = 0) :
    IRouteCommand {

    override fun navigate() {
        activity.finish()
        if (enterAnimation != 0 || exitAnimation != 0) {
            activity.overridePendingTransition(enterAnimation, exitAnimation)
        }
    }
}
